app/models/moment.rb

```diff
@@ -38,6 +38,15 @@ class Moment < ApplicationRecord
   validates :userid, :name, :why, presence: true
   validates :why, length: { minimum: 1, maximum: 2000 }
   validates :fix, length: { maximum: 2000 }
+  validates :secret_share_expires_at,
+            presence: true, if: :secret_share_identifier?
+
+  scope :find_secret_share!, lambda { |identifier|
```

Comment:
Personally, I would prefer to use a class method here instead of a `scope`.

```ruby
  def self.find_secret_share!(identifier)
    find_by!(
      'secret_share_expires_at > NOW()',
      secret_share_identifier: identifier
    )
  end
```

I know this is something that is a little controversial in the Rails community, but here are the reasons I think a class method is better:

- `scope` is not familiar to all Ruby/Rails developers. It's ActiveRecord "magic" that people have to learn, along with the `lambda` syntax, which is also unfamiliar to many newer Ruby developers.
- Class methods are used in a lot of places in Ruby code, so they are more familiar to most Ruby developers, and even new Ruby developers who have not learned them yet will encounter them before too long.
- The [docs for `scope`](http://api.rubyonrails.org/v5.1/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-scope) say "this is simply 'syntactic sugar' for defining an actual class method". In this case, however, the `scope` syntax is not really any shorter or more readable than the equivalent class method, so it doesn't provide much benefit.
- The docs also say, "The method is intended to return an ActiveRecord::Relation object, which is composable with other scopes." In other words, part of the purpose of defining a `scope` is that you can chain more criteria on to the end of it, but that is not possible here, because this performs a `find_by!` that returns the actual `Moment`, not a relation that can be chained.

Here are some good blog posts that describe the intended use of `scope`, and when it is better to use a `scope` than a class method and vice-versa:

- http://blog.plataformatec.com.br/2013/02/active-record-scopes-vs-class-methods/
- https://www.justinweiss.com/articles/should-you-use-scopes-or-class-methods/

Reply:
Julia are we making this a class method or keeping it as a scope?

Reply:
Gotta love when your coaches throw you around with conflicting information 😆

Reply:
@SophieRoseMcDonald while I could fight @TimMoore all day about the value of using scopes, his last point about it returning an `ActiveRecord::Relation` is completely correct and I accept it was my mistake to suggest it in this instance.

Reply:
Sorry if I confused things! I'm not totally opposed to scopes... I think those articles make some very good arguments for using it in specific cases. Anyway, as long as we agree about it here, it's all good and fighting won't be necessary 😄

Reply:
Such an intense debate lol. I did some bedtime reading on best practices for scopes vs. method (thanks @TimMoore!). Do we care about whether we return a null object vs. a blank relation? Scopes will return a blank relation in the event no record exists. With methods, we have to address returning nil. I think returning nil would be fine in this situation.

Tim raises really good points about readability though. We do want to be beginner friendly. What about this "compromise" that's mentioned in the last article Tim shared?
```
Even then, it’s still good to put your selecting, sorting, joining, and filtering code inside scopes. Then, use your scopes inside your class method. You’ll end up with a clearer class method, and scopes you can use throughout your app.
```
Anyways time for bed for me lol.

Reply:
@julianguyen in this case, it will raise an error if the record doesn't exist, which is what we want here